src/full/Agda/TypeChecking/Rules/Decl.hs

```diff
@@ -744,6 +744,12 @@ checkPragma r p =
             Function{} -> markStatic x
             _          -> typeError $ GenericError "STATIC directive only works on functions"
         A.InjectivePragma x -> markInjective x
+        A.NotProjectionLikePragma qn -> do
+          def <- getConstInfo qn
+          case theDef def of
+            it@Function{} ->
+              modifyGlobalDefinition qn $ \def -> def { theDef = it { funProjection = Nothing } }
```

Comment:
If I understand correctly, `NOT_PROJECTION_LIKE` wants to mark a function to _never_ be considered for projection-likeness.  But just setting `funProjection` to `Nothing` might not achieve this, because Agda might run the analysis for projection-likeness later (again) on that function. 

For `funTerminates` we have three valures:
- `Nothing`: termination check has not run yet
- `Just False`: non-terminating
- `Just True`: terminating

So the `TERMINATING` pragma will set `funTerminates` to `Just True`.

It seems you want to have these values for `funProjection`:
- `Right Projection`
- `Left MaybeProjectionLike`: projection-likeness check should still run
- `Left NotProjectionLike`: projection-likeness check should not be run